### PR TITLE
Updated 14.4.5, 14.4.6

### PR DESCRIPTION
**Pull request by Xmasreturns**
Improved readability, changed coercion example to involve less confusion
with respect to scoping (and reduce redundancy)

### CODE REVIEW

**Comment by steveklabnik** [comment]
Looks good to me, other than the Travis failure. @mdinger what do you think?

**Comment by Xmasreturns** [comment]
I swapped two lines that ended up with pointers to invalid data, so that should be reverted with this commit

File: examples/scope/lifetime/lifetime_bounds/bounds.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ fn print_ref<'a, T>(t: &'a T) where
 fn main() {
     let x = 7;
     let ref_x = Ref(&x);
-
+
```
Comment (mdinger) [inline examples/scope/lifetime/lifetime_bounds/bounds.rs]: I think this is kinda weird. I suspect the convention is blank lines are blank but am not sure

File: examples/scope/lifetime/lifetime_coercion/input.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
-A lifetime may be coerced into another if the the original is larger than
-the new lifetime:
+A longer lifetime can be coerced into a shorter one 
+so that it works inside a scope it normally wouldn't work in.
+This comes in the form of inferred coercion by the Rust compiler,
+and also in the form of declaring a lifetime difference:
```
Comment (mdinger) [inline examples/scope/lifetime/lifetime_coercion/input.md]: This sounds interesting and seems like a good place for a link to where this is elaborated on if such a place exists.

**Comment by mdinger** [comment]
I basically think it sounds good but I took only a very cursory look at it. If those two nits can be addressed or commented (or show what the actual convention is in the first case), I'd be fairly happy with this.

As wordsmithing, I'd definitely be happy with it. Regarding whether it addresses the problems highlighted in https://github.com/rust-lang/rust-by-example/issues/658, I'd have to do a more thorough review and comparison with niko's comment to see how they compare which I probably won't have time for until at least the weekend if not later. However, since this section isn't ideal as it is anyway, reevaluating this section sometime in the future after this was to merge wouldn't be too bad in my opinion.

Basically, I'm good with it.

File: examples/scope/lifetime/lifetime_bounds/bounds.rs (hunk 1 above)
Comment (steveklabnik) [inline examples/scope/lifetime/lifetime_bounds/bounds.rs]: Yes, blank lines should be blank

Comment (Xmasreturns) [inline examples/scope/lifetime/lifetime_bounds/bounds.rs]: This was unintentional, and will be changed.

File: examples/scope/lifetime/lifetime_coercion/input.md (hunk 2 above)
Comment (Xmasreturns) [inline examples/scope/lifetime/lifetime_coercion/input.md]: In this particular case, I'm letting the example elaborate on the claim.

As far as I can find, no single explanation currently exists that doesn't add confusion to a (relatively) simple lifetime explanation by talking about other aspects of lifetimes, cryptic compiler behaviors, or scopes. My solution to the issue was to combine a couple viewpoints from IRC and reword as succintly as possible.

**Comment by steveklabnik** [comment]
Sounds great to me then!

**Pull request merged**